### PR TITLE
add port check collector

### PR DESCRIPTION
**Pull request by yahonggu**

### CODE REVIEW

File: src/collectors/portcheck/portcheck.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+##### Dependencies
+
+* psutil
+
+Example config file PortCheckCollector.conf
+
+```
+enabled = True
+ttl = 120
```
Comment (TheConnMan) [inline src/collectors/portcheck/portcheck.py]: TTL should default to 150.

File: src/collectors/portcheck/portcheck.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+class PortCheckCollector(diamond.collector.Collector):
+
+    def __init__(self, *args, **kwargs):
+        super(PortCheckCollector, self).__init__(*args, **kwargs)
+        self.hostname = self.get_hostname()
+        self.ttl = self.config['ttl']
+
+        if not netuitive:
+            self.log.error('netuitive import failed. Heartbeat collector disabled')
```
Comment (TheConnMan) [inline src/collectors/portcheck/portcheck.py]: > Heartbeat collector

Copy/paste.

File: src/collectors/portcheck/portcheck.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+* psutil
+
+Example config file PortCheckCollector.conf
+
+```
+enabled = True
+ttl = 120
+
+[port]
+[[echo]]
```
Comment (TheConnMan) [inline src/collectors/portcheck/portcheck.py]: Is there any restriction on port name?

File: src/collectors/portcheck/portcheck.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+number = 8080
```
Comment (TheConnMan) [inline src/collectors/portcheck/portcheck.py]: Does this only support checking a local port, not a remote port?

Comment (yahonggu) [reply src/collectors/portcheck/portcheck.py]: right, only local tcp port for now.

File: src/collectors/portcheck/portcheck.py (hunk 3 above)
Comment (yahonggu) [reply src/collectors/portcheck/portcheck.py]: what restrictions are you thinking of? I didn't put any restriction now

File: src/collectors/portcheck/portcheck.py (hunk 4 above)
Comment (TheConnMan) [reply src/collectors/portcheck/portcheck.py]: The Nagios check referenced in the requirements allows for remote ports to be checked, you might want to confirm if this is a requirement or not: https://exchange.nagios.org/directory/Plugins/Network-Protocols/*-TCP-and-UDP-(Generic)/check_port-2Epl/details

Comment (yahonggu) [reply src/collectors/portcheck/portcheck.py]: confirmed this is not a requirement

**Review by TheConnMan — APPROVED** [review]

File: src/collectors/portcheck/portcheck.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    def collect(self):
+        """
+        Overrides the Collector.collect method
+        """
+
+        if psutil is None:
+            self.log.error('Unable to import module psutil')
+            return {}
+
+        for port_name, port_cfg in self.ports.iteritems():
```
Comment (andrewpaine) [inline src/collectors/portcheck/portcheck.py:100]: It looks like for each check you are getting all the port stats then searching for the port.
Would it be possible to make just one call to psutil.net_connections and reuse the results?

Comment (yahonggu) [reply src/collectors/portcheck/portcheck.py:100]: yeah, I inherit it from PortStatCollector. I can optimize even though I don't think there will be a lot of ports configured on each server.

**Review by andrewpaine — APPROVED** [review]

**Pull request merged**